pkg/dockerregistry/middleware/repository/openshift.go

```diff
@@ -188,9 +188,42 @@ func (r *repository) Put(ctx context.Context, manifest *manifest.SignedManifest)
 		},
 	}
 
-	if err := r.registryClient.ImageStreamMappings(r.namespace).Create(&irm); err != nil {
-		log.Errorf("Error creating ImageStreamMapping: %s", err)
-		return err
+	if err := r.registryClient.ImageStreamMappings(r.namespace).Create(&ism); err != nil {
```

Comment:
nit: in the future you might want to make the if block return and the else be the rest of the page (with a good comment about autoprovision).